CONTRIBUTING.md

```diff
@@ -24,9 +24,54 @@ Before sending a Pull Request with significant changes, please use the [issue tr
 OpenLineage uses [GitHub's fork and pull model](https://help.github.com/articles/about-collaborative-development-models/)
 to create a contribution.
 
-Every contribution is signed to say that the contributor has the rights to make the contribution and
+Every contribution is [signed](#sign-your-work) to say that the contributor has the rights to make the contribution and
```

Comment:
I would say "signed-off" here as this is different from cryptographically signing the commits.

We should link to https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md and remove the "Sign Your Work" section